holoviews/core/data.py

```diff
@@ -557,6 +557,19 @@ def array(cls, columns, dimensions):
         return Element.array(columns, dimensions)
 
     @classmethod
+    def recarray(cls, columns, dimensions):
+        if dimensions is None:
```

Comment:
Seems to be an odd method to introduce just for sorting purposes, especially as it is only needed to figure out a sort order (via ``argsort`` on the returned record array).

Generally I was hoping to avoid record arrays entirely and wouldn't want them to spread elsewhere. That said, the implementation here is fine and right now this method isn't used in a way that is likely to cause problems later. I don't yet understand why sorting record arrays is a simpler approach but I assume you wouldn't introduce them without a good reason!